### DISCUSSION TITLE
Safety of keeping a component's data pointer long term.

### DISCUSSION THREAD

**Discussion by GsLogiMaker** [SanderMertens/flecs#1145] [Q&A] [answered]
I'm keeping a mutable pointer to a component's data (which I got from `ecs_get_mut_id`) in an object long term. Is there anything that would invalidate the pointer, or otherwise make this unsafe besides deleting the component, entity, or world? Could adding or deleting entities or components cause the contents of the pointer to be move and are there any other problems I should look out for?

I couldn't find anything about this in the documentation.

**Comment by copygirl** [comment, marked as answer]
I think it would be wise to consider the pointer invalid as soon as you return control to Flecs. Besides modifying entities that share the same archetype (share the same components) and modifying the components on the entity itself, one thing that also comes to mind would be if you have any sorting queries. It *might* be safe if you can guarantee the entity's archetype never changes, but this sounds like a design flaw from the start. If what you're trying to achieve can't be done using ECS alone, what about storing a shared pointer in the component itself?

**Reply by GsLogiMaker** [reply]
I should be able to work around it; I was just hoping I wouldn't have to. I'm making a wrapper for Flecs so it can be used in Godot, and it was just easier to store the pointer in the component object.

Thanks for the info!
